### PR TITLE
Introduce xyBase in applicable series

### PR DESCRIPTION
**Pull request by chrisprice**
Also tidies up imports for all series.

### CODE REVIEW

File: tests/series/areaSpec.js [hunk 1]
```diff
@@ -18,12 +18,12 @@ describe('area', function() {
...
             .call(area);
 
-        expect(xValueSpy.calls.count()).toEqual(data.length);
-        this.utils.verifyAccessorCalls(xValueSpy, data);
-
-        // the defined call also invokes the y value accessors,
+        // the defined call also invokes the value accessors,
         // therefore they are invoked twice for each data point
 
+        expect(xValueSpy.calls.count()).toEqual(data.length * 2);
```
Comment (ColinEberhardt) [inline tests/series/areaSpec.js]: just wondering why the xValue accessor is now invoked twice for each point?

File: src/series/area.js [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
 
 export default function() {
 
-    var decorate = noop,
-        xScale = d3.time.scale(),
-        yScale = d3.scale.linear(),
-        y0Value = d3.functor(0),
-        y1Value = function(d, i) { return d.close; },
-        xValue = function(d, i) { return d.date; };
+    var decorate = noop;
```
Comment (ColinEberhardt) [inline src/series/area.js]: Any reason not to add `decorate` to the base component?

**Comment by ColinEberhardt** [comment]
A couple of minor comments, but otherwise LGTM

File: tests/series/areaSpec.js (hunk 1 above)
Comment (chrisprice) [inline tests/series/areaSpec.js]: As per the comment I changed defined to also check x/x0. It's a change to what we had but it seems the logical behaviour.

File: src/series/area.js (hunk 2 above)
Comment (chrisprice) [inline src/series/area.js]: It didn't feel like it belonged. I thought about putting it on its own component but that seemed overkill.

**Pull request merged**